packages/block-editor/src/components/global-styles/shadow-panel-components.js

```diff
@@ -123,3 +124,47 @@ function renderShadowToggle() {
 		);
 	};
 }
+
+function ClearShadowButton( { onClear } ) {
+	return (
+		<Button
+			className="block-editor-global-styles__clear-shadow"
+			variant="tertiary"
+			onClick={ onClear }
+		>
+			{ __( 'Clear' ) }
+		</Button>
+	);
+}
+
+export function getShadowPresets( settings ) {
+	if ( ! settings?.shadow ) {
+		return [];
+	}
+
+	const defaultPresetsEnabled = settings?.shadow?.defaultPresets;
+	const defaultShadows = defaultPresetsEnabled
+		? settings?.shadow?.presets?.default || []
+		: [];
+	const customShadows = settings?.shadow?.presets?.custom || [];
+	const themeShadows = settings?.shadow?.presets?.theme || [];
+
+	// show unset option only in global styles
+	const isGlobalStyles = settings?.hasOwnProperty( 'appearanceTools' );
```

Comment:
I can see that settings in global styles always comes with `appearanceTools` set to false and in block style settings, this key is not set. It is working as per [this](https://github.com/WordPress/gutenberg/pull/58766#issuecomment-1942015584) expectation. 

Need feedback on this condition.

Reply:
@t-hamano could you share feedback on this.